packages/amplify-category-interactions/src/provider-utils/awscloudformation/service-walkthroughs/lex-walkthrough.js

```diff
@@ -346,84 +271,62 @@ async function configure(context, defaultValuesFilename, serviceMetadata, resour
-  intentName = intentName[inputs[12].key];
+async function addIntent(context, resourceName, intents, parameters) {
+  let intentName = await askIntent(resourceName);
 
   // Checks for duplicate intent names
   while (
     intents.filter(intent => intent.intentName === intentName).length > 0 ||
     (parameters && parameters.intents.filter(intent => intent.intentName === intentName).length > 0)
   ) {
-    print.info('');
-    print.info('Intent names must be unique');
-    print.info('');
-    intentName = await inquirer.prompt(intentNameQuestion);
-    intentName = intentName[inputs[12].key];
+    printer.blankLine();
+    printer.info('Intent names must be unique');
+    printer.blankLine();
+    intentName = await askIntent(resourceName);
   }
 
-  const utterances = await addUtterance(context, intentName, botName, resourceName, serviceMetadata);
+  const utterances = await addUtterance(resourceName);
 
-  print.info('');
-  print.info('Now, add a slot to your intent. A slot is data the user must provide to fulfill the intent.');
-  print.info('');
+  printer.blankLine();
+  printer.info('Now, add a slot to your intent. A slot is data the user must provide to fulfill the intent.');
+  printer.blankLine();
 
   let slots = [];
   let newSlotTypes = [];
-  const slotReturn = await addSlot(context, intentName, botName, resourceName, serviceMetadata, parameters);
+  const slotReturn = await addSlot(context, intentName, resourceName, parameters);
   if (slotReturn.length > 1) {
     newSlotTypes = slotReturn[1];
   }
   slots = slotReturn[0];
 
-  const addConfirmationQuestion = {
-    type: inputs[18].type,
-    name: inputs[18].key,
-    message: inputs[18].question,
-    default: inputs[18].default,
-  };
   let confirmationQuestion;
   let cancelMessage;
-  const addConfirmation = await inquirer.prompt(addConfirmationQuestion);
-  if (addConfirmation[inputs[18].key]) {
-    const confirmationQuestionQuestion = {
-      type: inputs[19].type,
-      name: inputs[19].key,
-      message: inputs[19].question,
-      validate: amplify.inputValidation(inputs[19]),
-    };
-    confirmationQuestion = await inquirer.prompt(confirmationQuestionQuestion);
-    confirmationQuestion = confirmationQuestion[inputs[19].key];
-
-    const cancelMessageQuestion = {
-      type: inputs[20].type,
-      name: inputs[20].key,
-      message: inputs[20].question,
-      validate: amplify.inputValidation(inputs[20]),
-    };
-    cancelMessage = await inquirer.prompt(cancelMessageQuestion);
-    cancelMessage = cancelMessage[inputs[20].key];
+  if (await prompter.yesOrNo('Would you like to add a confirmation prompt to your intent?', false)) {
+    confirmationQuestion = await prompter.input(
+      'Enter a confirmation message (e.g. Are you sure you want to order a {Drink_name}?):',
+      {
+        validate: matchRegex(/^.{1,1000}$/, 'Confirmation questions can have a maximum of 1000 characters and cannot be empty'),
```

Comment:
Nit: validators like this one might be slightly more readable as:
```
and(minLength(1), maxLength(1000), 'Confirmation questions can have a maximum of 1000 characters and cannot be empty')
```

Reply:
updated